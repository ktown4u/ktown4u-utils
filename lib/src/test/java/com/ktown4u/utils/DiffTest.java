package com.ktown4u.utils;

import org.approvaltests.Approvals;
import org.junit.jupiter.api.DisplayName;
import org.junit.jupiter.api.Test;

public class DiffTest {
    @Test
    @DisplayName("두 문자열을 비교한다")
    void diff() {
        final String before = before();
        final String after = after();

        Approvals.verify(Diff.between(before, after));
    }

    private String after() {
        return """
                totalPrice: 100
                totalDiscountPrice: 10
                finalPrice: 90
                mileageToBeEarned: 0
                totalQty: 2
                currency: "USD"
                tube:
                  name: ""
                  price: 0
                  qty: 0
                  active: false
                items:
                - productId:
                    shopNo: 164
                    productNo: 1
                    bundleNo: 2
                    fanClubProductNo: 3
                    eventNo: 4
                  qty: 2
                  name: "name"
                  image: "image"
                  retailPrice: 100
                  sellPrice: 90
                  artist: "artist"
                  brand: "brand"
                  saleStatusCode: "ON_SALE"
                  isAdultOnly: true
                  additionalItems: []
                """;
    }

    private String before() {
        return """
                totalPrice: 0
                totalDiscountPrice: 0
                finalPrice: 0
                mileageToBeEarned: 0
                totalQty: 0
                currency: "USD"
                tube:
                  name: ""
                  price: 0
                  qty: 0
                  active: false
                items: []
                """;
    }

}
