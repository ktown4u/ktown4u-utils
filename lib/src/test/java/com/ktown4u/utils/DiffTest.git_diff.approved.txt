- totalPrice: 0
+ totalPrice: 100
- totalDiscountPrice: 0
+ totalDiscountPrice: 10
- finalPrice: 0
+ finalPrice: 90
mileageToBeEarned: 0
- totalQty: 0
+ totalQty: 2
currency: "USD"
tube:
  name: ""
  price: 0
  qty: 0
  active: false
- items: []
+ items:
+ - productId:
+     shopNo: 164
+     productNo: 1
+     bundleNo: 2
+     fanClubProductNo: 3
+     eventNo: 4
+   qty: 2
+   name: "name"
+   image: "image"
+   retailPrice: 100
+   sellPrice: 90
+   artist: "artist"
+   brand: "brand"
+   saleStatusCode: "ON_SALE"
+   isAdultOnly: true
+   additionalItems: []
